sys/ssp/ssp.c

```diff
@@ -22,9 +22,6 @@
 
 #include "panic.h"
 
-/* this should be randomized for each build */
-#define STACK_CHK_GUARD 0x595e9fbd94fda766
```

Comment:
I suggest to make this `#ifndef ..., #define ..., #endif`.

Reply:
To use a hardcoded value as fallback for the CI / reproducible builds?

Reply:
Yeah.

Reply:
I would prefer doing this in Makefile.include to reduce the likelihood of accidentally using a hardcoded canary value. For example, changing `Makefile.include` to:

```Makefile
 # Generate a canary value for the ssp module
ifneq (,$(RIOT_CI_BUILD))
RIOTCANARY := $(shell $(RIOTTOOLS)/randhex/randhex.sh)
else
RIOTCANARY := 0x23422342
endif
```

Would that work for you?

Reply:
ok!

Reply:
Alright, just pushed a change implementing this behaviour.